src/Apitizer/GenericApi/AuthorizedController.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Apitizer\GenericApi;
+
+use Apitizer\QueryBuilder;
+use Illuminate\Foundation\Auth\Access\AuthorizesRequests;
+
+class AuthorizedController extends Controller
+{
+    use AuthorizesRequests;
+
+    protected function setup(): void
+    {
+        parent::setup();
+
+        $this->authorizeResource($this->queryBuilder->model());
+    }
```

Comment:
Looking back at it now, this probably won't work, as the authorizeResource is implemented as middleware and must therefore be implemented in the constructor. However, due to the way the schema is smuggled into the controller, we won't yet know which schema it is in the constructor. So we'll have to reimplement these as separate `authorize` calls in the controller methods.